EmbeddedResourceInjectorTask error in Visual Studio

### Current behavior

After updating to V.S. 17.14.0 Prev-7 I'm getting this error when I try to execute the app (F5):

Project uses Uno.Sdk 5.6.51 and only targets Desktop.

```
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018: Error inesperado en la tarea "EmbeddedResourceInjectorTask_vb6e37c52d13e4ace2ae084320c1fecbbaa7fabe4".
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018: System.Exception->Microsoft.Build.Framework.BuildException.GenericBuildTransferredException: Mono.Cecil.Cil.SymbolsNotFoundException: No symbol found for file: obj\Release\net9.0-desktop\osmoui.dll
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018:    en Mono.Cecil.Cil.DefaultSymbolReaderProvider.GetSymbolReader(ModuleDefinition module, String fileName)
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018:    en Mono.Cecil.ModuleReader.ReadSymbols(ModuleDefinition module, ReaderParameters parameters)
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018:    en Mono.Cecil.ModuleReader.CreateModule(Image image, ReaderParameters parameters)
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018:    en Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
3>C:\Users\joan\.nuget\packages\uno.winui\5.6.91\buildTransitive\Uno.UI.Tasks.targets(290,3): error MSB4018:    en Uno.UI.Tasks.EmbeddedResourceInjector.EmbeddedResourceInjectorTask_vb6e37c52d13e4ace2ae084320c1fecbbaa7fabe4.Execute() en /agent/_work/1/s/src/SourceGenerators/Uno.UI.Tasks/EmbeddedResourceInjector/EmbeddedResourceInjectorTask.cs:línea 61
```
I can run the app using:
```
dotnet run -c Debug -f net9.0-desktop
```

I get the same error when I try to publish the application to a network share or directly to the a linux terminal using this command:

```
dotnet publish -r linux-arm64 -c Release -f net9.0-desktop --self-contained true -o S:\UI  /p:PublishSingleFile=true /p:DebugSymbols=false /p:DebugType=None
```

But if I run the command twice, the second time it works.

I've tried with vscode-insiders and it runs with no errors.


### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

I can't reproduce it by now.

### Workaround

Using `dotnet run -c Debug -f net9.0-desktop`

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Renderer

- [x] Skia
- [ ] Native

### Affected platforms

Desktop (Windows)

### IDE

Visual Studio 2022

### IDE version

17.14.0 Prev 7

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

--- a/build/test-scripts/run-netcore-mobile-template-tests.ps1
+++ b/build/test-scripts/run-netcore-mobile-template-tests.ps1
@@ -312,6 +312,9 @@ $projects =
     # Publishing validation
     @(4, "5.3/uno53net9blank/uno53net9blank/uno53net9blank.csproj", @("-f", "net9.0-desktop", "-p:TargetFrameworks=net9.0-desktop", "-p:PackageFormat=app", "-r", "osx-x64"), @("OnlyMacOS", "NetCore", "Publish")),
 
+    # Publish with no debug symbols validation
+    @(4, "5.3/uno53net9blank/uno53net9blank/uno53net9blank.csproj", @("-f", "net9.0-desktop", "-p:TargetFrameworks=net9.0-desktop", "-r", "win-x64", "-p:DebugSymbols=false", "-p:DebugType=None"), @("NetCore", "Publish")),
+
     # Ensure that build can happen even if a RID is specified
     @(4, "5.3/uno53net9blank/uno53net9blank/uno53net9blank.csproj", @("-f", "net9.0-android", "-r", "android-arm64"), @("macOS", "NetCore"))
 

--- a/src/SourceGenerators/Uno.UI.Tasks/EmbeddedResourceInjector/EmbeddedResourceInjectorTask.cs
+++ b/src/SourceGenerators/Uno.UI.Tasks/EmbeddedResourceInjector/EmbeddedResourceInjectorTask.cs
@@ -16,6 +16,7 @@ using System.Xml;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 using Mono.Cecil;
+using Mono.Cecil.Cil;
 
 namespace Uno.UI.Tasks.EmbeddedResourceInjector
 {
@@ -58,7 +59,13 @@ namespace Uno.UI.Tasks.EmbeddedResourceInjector
 						resolver.AddSearchDirectory(path);
 					}
 
-					using (var asm = AssemblyDefinition.ReadAssembly(TargetAssembly, new ReaderParameters() { AssemblyResolver = resolver, ReadSymbols = true, ReadWrite = true }))
+					using (var asm = AssemblyDefinition.ReadAssembly(TargetAssembly, new()
+					{
+						AssemblyResolver = resolver,
+						ReadSymbols = true,
+						ReadWrite = true,
+						SymbolReaderProvider = new DefaultSymbolReaderProvider(throwIfNoSymbol: false),
+					}))
 					{
 						foreach (var embeddedResource in EmbeddedResources)
 						{

--- a/src/SourceGenerators/Uno.UI.Tasks/EmbeddedResourceInjector/EmbeddedResourceInjectorTask.cs
+++ b/src/SourceGenerators/Uno.UI.Tasks/EmbeddedResourceInjector/EmbeddedResourceInjectorTask.cs
@@ -88,7 +88,7 @@ namespace Uno.UI.Tasks.EmbeddedResourceInjector
 							asm.MainModule.Resources.Add(new EmbeddedResource(logicalName, ManifestResourceAttributes.Public, File.ReadAllBytes(embeddedResource.GetMetadata("FullPath"))));
 						}
 
-						asm.Write(new WriterParameters() { WriteSymbols = true });
+						asm.Write(new WriterParameters() { WriteSymbols = asm.MainModule.SymbolReader is not null });
 					}
 
 					WaitForUnlockedFile(TargetAssembly);

